### PR TITLE
[WIP] Fullpdf

### PR DESCRIPTION
**Pull request by voisardf**
This PR implements the full PDF as specified by the confederation and other style improvments

### CODE REVIEW

File: crdppf/lib/content.py [hunk 1]
```diff
@@ -571,11 +542,14 @@ def get_content(id, request):
...
+
     # import json
-    # # pretty printed json data for the extract
+    # pretty printed json data for the extract
     # jsonfile = open('C:/Temp/extractdata.json', 'w')
     # jsondata = json.dumps(d, indent=4)
     # jsonfile.write(jsondata)
     # jsonfile.close()
-
+    # sdf
```
Comment (kalbermattenm) [inline crdppf/lib/content.py]: I would remove this line...

File: crdppf/views/printproxy.py [hunk 2]
```diff
@@ -87,17 +100,45 @@ def report_create(self):
...
+            out = tempfile.NamedTemporaryFile(suffix='.pdf')
+            cmd += ['cat', 'output', out.name]
+            sys.stdout.flush()
+            time.sleep(0.1)
+            subprocess.check_call(cmd)
+            print_result.body = out.file.read()
+            result = print_result
+        else:
+            result = print_result
+            #~ content = print_result.body
```
Comment (kalbermattenm) [inline crdppf/views/printproxy.py]: Please remove

File: crdppf/views/printproxy.py [hunk 3]
```diff
@@ -87,17 +100,45 @@ def report_create(self):
...
+            main.file.write(print_result.body)
+            main.flush()
+            cmd = ['pdftk', main.name]
+            temp_files = [main]
+            for url in pdfs_to_join:
+                tmp_file = tempfile.NamedTemporaryFile(suffix='.pdf')
+                result = requests.get(url)
+                tmp_file.write(result.content)
+                #~ pdf = urlopen(Request(url)).read()
+                #~ tmp_file.write(pdf)
```
Comment (kalbermattenm) [inline crdppf/views/printproxy.py]: Remove these two lines

File: crdppf/views/printproxy.py [hunk 4]
```diff
@@ -87,17 +100,45 @@ def report_create(self):
...
+                tmp_file.flush()
+                temp_files.append(tmp_file)
+                cmd.append(tmp_file.name)
+            out = tempfile.NamedTemporaryFile(suffix='.pdf')
+            cmd += ['cat', 'output', out.name]
```
Comment (kalbermattenm) [inline crdppf/views/printproxy.py]: What is this command `cat` ?? Is it some kind of PDF aggregator ?

If yes, you should describe it as a mandatory soft to install + define in the path in the requirements...

File: setup.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
         'papyrus',
         'GeoAlchemy2',
         'OWSLib',
-        'fpdf',
+        'pypdftk',
```
Comment (kalbermattenm) [inline setup.py]: Is it used ???

I saw that you rather use a subprocess, thus I do not get why you need pypdftk (by the way, why not use this library???)

File: crdppf/lib/content.py [hunk 6]
```diff
@@ -74,54 +73,19 @@ def set_documents(request, topicid, doctype, docids, featureinfo, geofilter, top
...
-                documents.append({"officialtitle": doc['officialtitle'], "title": doc['title'], "remoteurl": doc['remoteurl']})
-                if doc['doctype'] != u'legalbase' and doc['documentid'] not in doclist:
-                    appendices.append(add_appendix(topicid, 'A'+str(len(appendices)+1), unicode(doc['officialtitle']).encode('iso-8859-1'),
-                                                   unicode(doc['title']).encode('iso-8859-1'), unicode(
-                        doc['remoteurl']).encode('iso-8859-1'), doc['localurl'], topicdata))
-                if doc['documentid'] not in doclist:
-                    doclist.append(doc)
-    if documents == []:
-        documents = [{"officialtitle": "", "title": "", "remoteurl": ""}]
+                documents.append({"documentid": doc['documentid'], "officialtitle": doc['officialtitle'], "title": doc['title'], "remoteurl": doc['remoteurl']})
```
[the remark this answers is not in the document]
Comment (voisardf) [reply crdppf/lib/content.py]: almost, but not the same condition :)

File: crdppf/lib/content.py (hunk 1 above)
Comment (voisardf) [reply crdppf/lib/content.py]: I will for merging :)

File: crdppf/lib/content.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (kalbermattenm) [reply crdppf/lib/content.py]: Oh... OK, I missed line 84...

**Comment by voisardf** [comment]
Checked on prepub instance, merging

**Pull request merged**